File: backend-spring-boot/src/main/java/no/vicx/backend/jwt/github/vm/GitHubUserVm.java
package no.vicx.backend.jwt.github.vm;

import com.fasterxml.jackson.annotation.JsonProperty;

public record GitHubUserVm(
        String id,
        String login,
        String name,
        String email,
        @JsonProperty("avatar_url")
        String avatarUrl
) {
        public boolean isEmpty(){
                return id == null
                        && login == null
                        && name == null
                        && email == null
                        && avatarUrl == null;
        }
}

File: backend-spring-boot/src/main/kotlin/no/vicx/backend/jwt/github/vm/GitHubUserVm.kt
package no.vicx.backend.jwt.github.vm

import com.fasterxml.jackson.annotation.JsonProperty

data class GitHubUserVm(
    val id: String = "",
    val login: String = "",
    val name: String = "",
    val email: String = "",
    @JsonProperty("avatar_url") val avatarUrl: String = "" // TODO: Check if this should be init with null
) {
    val isEmpty: Boolean
        get() = id.isBlank()
                && login.isBlank()
                && name.isBlank()
                && email.isBlank()
                && avatarUrl.isBlank()
}
